### PR TITLE
custom 404 page

### PR DESCRIPTION
**Pull request by SanaBob**

### CODE REVIEW

File: src/app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const Custom404 = ({ message }: { message: string }) => {
+  return (
+    <>
```
Comment (Opeyem1a) [inline src/app/not-found.tsx:3]: what does this look like? Just because it's a 404 page, doesn't mean we should drop our standards for visual polish

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/app/not-found.tsx (hunk 1 above)
Comment (SanaBob) [reply src/app/not-found.tsx]: Screenshot 2024-04-29 at 11 26 57 AM
It looks like this now, let me know what you think

File: src/app/not-found.tsx
Comment (Opeyem1a) [inline src/app/not-found.tsx:1]: is naming this file `not-found` standard practice? I thought naming it `404.tsx` was the standard in nextjs

File: src/app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Text } from "@/components/ui/text"
+
+const Custom404 = ({ errorMessage }: { errorMessage: string }) => {
+  return (
+    <div className="flex h-screen">
+      <div className = "m-auto flex flex-col items-center">
```
Comment (Opeyem1a) [inline src/app/not-found.tsx:6]: pls add more space between the text here

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 3]
```diff
@@ -40,6 +41,10 @@ const useCourseProvider = (): CourseContextType => {
 
 export const CourseProvider: React.FC<PropsWithChildren> = ({ children }) => {
   const courseContext = useCourseProvider()
+  if (!courseContext.courseId) {
+    return Custom404({ errorMessage: "Course not found" })
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:45]: It feels weird to me that we are not calling this function in the `<Component />` way

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/app/not-found.tsx
Comment (SanaBob) [reply src/app/not-found.tsx:1]: This was the case but it was changed in next 14.0

Comment (Opeyem1a) [reply src/app/not-found.tsx:1]: sounds good, thanks for checking

File: src/app/not-found.tsx (hunk 2 above)
Comment (Opeyem1a) [reply src/app/not-found.tsx]: it is a nit, but I do think the title is crunched too close to the error message text

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**